### PR TITLE
Added support for cgroup parent

### PR DESCRIPTION
**Pull request by mohitsoni**
This change adds cgroup-parent support to docker-py, which will allow us to add cgroup-parent support inside compose.

### CODE REVIEW

**Comment by aanand** [comment]
LGTM

File: docker/client.py [hunk 1]
```diff
@@ -737,7 +738,7 @@ def start(self, container, binds=None, port_bindings=None, lxc_conf=None,
               dns=None, dns_search=None, volumes_from=None, network_mode=None,
               restart_policy=None, cap_add=None, cap_drop=None, devices=None,
               extra_hosts=None, read_only=None, pid_mode=None, ipc_mode=None,
-              security_opt=None, ulimits=None):
+              security_opt=None, ulimits=None, cgroup_parent=None):
```
Comment (shin-) [inline docker/client.py]: I'd argue against adding this - we've deprecated passing host config arguments in `start`, and will probably remove it entirely in a future version.

Comment (mohitsoni) [inline docker/client.py]: @shin- Good to know. Can you please tell how ```cgroup_parent``` and other host config arguments be passed ?

Comment (shin-) [inline docker/client.py]: Gladly! You can see the [documentation](http://docker-py.readthedocs.org/en/latest/hostconfig/) on that topic.

**Comment by shin-** [comment]
Applied a few fixes and merged manually. Thanks!

**Pull request closed without merging**